compareBuckets/compareBuckets.js

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable max-len */
+/* eslint-disable no-console */
+/* eslint-disable comma-dangle */
+
+const async = require('async');
+
+const { listBucketMasterKeys, getObjectURL } = require('./utils');
+
+function compareBuckets(params, log, cb) {
+    const {
+        bucketdSrcParams,
+        bucketdDstParams,
+        statusObj,
+    } = params;
+
+    statusObj.srcBucketInProgress = bucketdSrcParams.bucket;
+    statusObj.dstBucketInProgress = bucketdDstParams.bucket;
+
+    let srcDone = false;
+    let dstDone = false;
+    let srcMarker = '';
+    let dstMarker = '';
+    let srcContents = [];
+    let dstContents = [];
+
+    async.doWhilst(
+        done => {
+            async.parallel({
+                src: _done => {
+                    if (srcDone || srcContents.length > 0) {
+                        return process.nextTick(_done);
+                    }
+
+                    return listBucketMasterKeys(bucketdSrcParams,
+                        (err, isTruncated, marker, contents) => {
+                            if (err) {
+                                return _done(err);
+                            }
+
+                            srcContents = contents;
+                            srcDone = !isTruncated;
+                            srcMarker = marker;
+                            statusObj.srcKeyMarker = srcMarker;
```

Comment:
Here `bucketdSrcParams.marker` should be updated, fixed it to avoid looping over the same range

Reply:
thanks! didn't catch this with the mock.